politeiad/cmd/legacypoliteia/importcmd.go

```diff
@@ -134,24 +141,240 @@ func importProposals(legacyDir string, cmd *importCmd) error {
 //
 // This function assumes that the proposal does not yet exist in tstore.
 // Handling proposals that have been partially added is done by the
-// fsckPropsal() function.
+// fsckProposal() function.
+//
+// The steps this function executes to accomplish the import are the following:
+//
+// 1. Create a new tlog tree for the legacy proposal.
 //
-// This function replaces the git backend token with the tstore backend token
-// in the following structures:
-// - Fill
-// - In
-// - This
-// - List
+// 2. Replace the git backend tokens with the tstore backend token in the
+//    following structures:
+//      - Record Metadata
+//      - Vote Details Params
+//      - Vote Metadata LinkTo
 //
-// The git token to tstore token mapping is added to the memory cache for all
-// RFP parent proposals.
+// 3. Saves the auth details blob and vote details blob if they exist for
+//    the legacy proposal being imported.
 //
-// The parent token of all RFP submissions are updated with the tstore RFP
-// parent token.
-func importProposal(legacyDir, gitToken string, cmd *importCmd) {}
+// 4. Import comment blobs and vote blobs concurrently. This is the most costly
+//    process of importing, and is why it is done concurrently.
+//
+// 5. Add git token to tstore token mapping to the memory cache for all RFP
+// 	  parent proposals. The parent token of all RFP submissions are updated
+//    with the tstore RFP parent token.
+func importProposal(legacyDir, gitToken string, cmd *importCmd) error {
```

Comment:
This function needs to be easier to read and review.  I don't want to have to go searching into nested functions to find when something is being saved.

https://github.com/decred/politeia/blob/fa7b7ad91df5b1ee23a26e1358d950dd222e4dbf/politeiad/cmd/legacypoliteia/convertcmd.go#L94-L176

In the above code, each function corresponds to a field on the proposal and they are called in the same order that the fields are listed in the proposal.  I can easily go down and audit this code.  If I want to see exactly how one of the fields is being derived, I can look into the convert function for further details.

Reply:
It  would also help with readability if you used the naming convention `import` and put all of the import functions into a `import.go` file.

Ex. `importRecordMetadata()`, `importFiles()`, etc